langchain4j-vertex-ai/src/main/java/dev/langchain4j/model/vertexai/VertexAiMatchingEngine.java

```diff
@@ -0,0 +1,290 @@
+        if (json != null) {
+            final VertexAiEmbeddingIndex embeddingIndex = VertexAiEmbeddingIndex.fromJson(json);
+            final List<String> indicesToDelete = embeddingIndex
+                    .getRecords()
+                    .stream()
+                    .map(VertexAiEmbeddingIndexRecord::getId)
+                    .collect(Collectors.toList());
+
+            // Delete documents
+            indicesToDelete.forEach(id -> getGcpBlobService().delete("documents/" + id));
+
+            // Delete index
+            getIndexEndpointService().deleteIndices(indicesToDelete);
+        }
+
+        // Delete index file
+        getGcpBlobService().delete(path);
+    }
+
+    /**
+     * Get all indices.
+     *
+     * @return the list of indices
+     */
+    public List<String> allIndices() {
+        final String prefix = "indexes/";
+
+        return getGcpBlobService()
+                .list()
+                .filter(name -> name.startsWith(prefix))
+                .map(name -> name.substring(prefix.length(), name.length() - 5))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Initializes the storage.
+     *
+     * @return the storage
+     */
+    private GcpBlobService initBlob() {
+        ensureNotNull(bucketName, "bucketName is null");
+        ensureNotNull(project, "project is null");
+
+        return GcpBlobService.builder()
+                .bucketName(bucketName)
+                .credentialsProvider(credentialsProvider)
+                .project(project)
+                .build();
+    }
+
+    /**
+     * Initializes the matching service.
+     *
+     * @return the matching service
+     */
+    private MatchingService initMatchingService() {
+        ensureNotNull(deployedIndexId, "deployedIndexId is null");
+        ensureNotNull(indexId, "indexId is null");
+        ensureNotNull(indexEndpointId, "indexEndpointId is null");
+        ensureNotNull(project, "project is null");
+        ensureNotNull(location, "location is null");
+
+        return MatchingService.builder()
+                .gcpBlobService(getGcpBlobService())
+                .deployedIndexId(deployedIndexId)
+                .credentialsProvider(credentialsProvider)
+                .returnFullDatapoint(returnFullDatapoint)
+                .indexEndpointId(indexEndpointId)
+                .indexId(indexId)
+                .project(project)
+                .location(location)
+                .endpoint(getIndexEndpointService().getPublicEndpoint())
+                .build();
+    }
+
+    /**
+     * Initializes the index endpoint.
+     *
+     * @return the index endpoint
+     */
+    private IndexEndpointService initIndexEndpoint() {
+        ensureNotNull(endpoint, "endpoint is null");
+        ensureNotNull(location, "location is null");
+        ensureNotNull(project, "project is null");
+
+        return IndexEndpointService
+                .builder()
+                .endpoint(endpoint)
```

Comment:
is is supposed to be used like this: `.endpoint(ensureNotNull(endpoint, "endpoint"))...`

Reply:
Coding style, not worries if you like it better this way

Reply:
Done.